File: frontend/src/api/comment.ts
import { api } from './client'

export interface Comment {
  post_id: number
  comment_text: string
  creator_username: string
  creator_id: number
}

export async function createComment(postId: number, comment_text: string): Promise<{ comment_id: number }> {
  const { data } = await api.post<{ comment_id: number }>(`/comment/create/${postId}`, {
    comment_text,
  })
  return data
}

export async function getCommentsByPostId(postId: number): Promise<Comment[]> {
  const { data } = await api.get<Comment[]>(`/comment/all/${postId}`)
  return data
}

File: backend/internal/domain/comment.go
package domain

import "myapp/internal/delivery/http/dto"

type Comment struct {
	ID              int    `db:"id"`
	CommentText     string `db:"text"`
	PostID          int    `db:"post_id"`
	CreatorID       int    `db:"creator_id"`
	CreatorUsername string `db:"creator_username"`
}

func (c Comment) ToCommentDTO() dto.CommentDTO {
	return dto.CommentDTO{
		CommentID:       c.ID,
		PostID:          c.PostID,
		CommentText:     c.CommentText,
		CreatorUsername: c.CreatorUsername,
		CreatorID:       c.CreatorID,
	}
}
